orchagent/fgnhgorch.cpp

```diff
@@ -204,29 +219,45 @@ void FgNhgOrch::setStateDbRouteEntry(const IpPrefix &ipPrefix, uint32_t index, N
     // Write to StateDb
     std::vector<FieldValueTuple> fvs;
 
-    // check if profile already exists - if yes - skip creation
+    // get the existing buckets from warm restart state table
```

Comment:
There is another PR which is attempting to solve the state db key sequencing problem. https://github.com/sonic-net/sonic-swss/pull/3566. 

I suggested a change there to modify the code to use hset instead of get + set as it is currently implemented. I think that hset is a simpler approach compared to creating a different data structure to track the index of the element inside redis(as is done here). Please try that and see if it works. In any case I have suggested the same change on the other PR linked above which is trying to solve this problem as well.